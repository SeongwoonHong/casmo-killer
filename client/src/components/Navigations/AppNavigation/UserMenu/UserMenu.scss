.User-menu {
    right: 0;
    transform: translateX(100%);
    @media screen and (min-width: 992px) {
        transform: translateX(0);
    }
}
