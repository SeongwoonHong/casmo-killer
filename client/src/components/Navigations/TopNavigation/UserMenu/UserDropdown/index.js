import { connect } from 'react-redux';
import UserMenuDropdown from './UserMenuDropdown';

import * as actions from '../../../../../actions';

const mapStateToProps = (state) => {
  return {
    layout: state.layout
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    logout: () => dispatch(actions.logout())
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(UserMenuDropdown);
