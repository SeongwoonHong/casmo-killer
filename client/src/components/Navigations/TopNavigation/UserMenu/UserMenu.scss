.user-menu {
    > .notification {
        display: none;
    }
    > .user-profile {
        cursor: pointer;
        &:active,
        &:focus {
            background-color: palette(Teal, 800);
            outline: none;
        }
        > .user-info {
            display: none;
        }
        > .dropdown-toggle {
            padding: 0;
            > .dd-toggler {
                width: 50px;
            }
        }
    }
    @media screen and (min-width: 768px) {
        @include flex-box(space-between);
        > .notification {
            display: block;
            margin-right: 10px;
            > .notification-badge {
                right: 0;
                top: 4px;
            }
        }
    }
    @media screen and (min-width: 992px) {
        > .user-profile {
            display: block;
            &.loggedIn {
                @include flex-box(space-between);
            }
            > .user-info {
                @include flex-box(flex-start);
                height: 50px;
                width: 210px;
                > .user-avatar {
                    background-color: transparent;
                    margin: 0 10px;
                    overflow: hidden;
                    > img {
                        @include rectanglify(34px);
                        border-radius: 50%;
                        display: block;
                    }
                }
                > .user-username {
                    color: palette(Teal, 50);
                    > span {
                        display: block;
                        line-height: normal;
                        &:first-child {
                            font-size: 14px;
                        }
                        &:last-child {
                            color: palette(Teal, 300);
                            font-size: 12px;
                        }
                    }
                }
            }
            > .dropdown-toggle {
                float: right;
            }
        }
    }
}
