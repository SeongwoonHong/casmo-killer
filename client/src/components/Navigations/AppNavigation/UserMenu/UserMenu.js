/* eslint no-unused-vars: 0 */
import React, { Component } from 'react';
import onClickOutside from 'react-onclickoutside';

import './UserMenu.scss';

import MenuContainer from '../shared/MenuContainer';

import UserDropdownBtn from './UserDropdownBtn';
import UserDropdown from './UserDropdown';

class UserMenu extends Component {

  constructor(props) {
    super(props);
    this.handleClickOutside = this.handleClickOutside.bind(this);
  }

  handleClickOutside(e) {
    if (this.props.active) {
      this.props.toggleUserMenu(false);
    }
  }

  render() {

    const {
      active, user, currentRoute, toggleUserMenu, logout
    } = this.props;

    return (
      <MenuContainer
        active={ active }
        className="User-menu"
        onClose={ () => toggleUserMenu(false) }>
        <UserDropdownBtn
          active={ active }
          isLoggedIn={ user.isLoggedIn }
          displayName={ user.displayName }
          avatar={ user.avatar }
          level={ user.level }
          toggleUserMenu={ toggleUserMenu } />
        <UserDropdown
          active={ active }
          isLoggedIn={ user.isLoggedIn}
          currentRoute={ currentRoute }
          onLogout={ logout }
          toggleUserMenu={ toggleUserMenu } />
      </MenuContainer>
    );

  }

}

export default onClickOutside(UserMenu);
