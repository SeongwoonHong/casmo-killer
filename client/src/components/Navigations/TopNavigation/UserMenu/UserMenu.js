/* eslint react/no-find-dom-node: 0 */
import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import { Link } from 'react-router-dom';
import classnames from 'classnames';
import './UserMenu.scss';

class UserMenu extends Component {

  constructor(props) {

    super(props);

    this.toggleDropdown = this.toggleDropdown.bind(this);
    this.clickToToggle = this.clickToToggle.bind(this);

  }

  componentDidMount() {

    document.addEventListener('click', this.toggleDropdown, false);
    document.addEventListener('touchend', this.toggleDropdown, false);

  }

  componentWillUnmount() {

    document.removeEventListener('click', this.toggleDropdown, false);
    document.removeEventListener('touchend', this.toggleDropdown, false);

  }

  toggleDropdown(event) {

    if (!ReactDOM.findDOMNode(this).contains(event.target)) {
      if (this.props.layout.isUserDropdownOpen) {
        setTimeout(() => {
          this.props.toggleUserDropdown(false);
        });
      }
    }

  }

  clickToToggle(event) {

    if (event.type === 'mousedown' && event.button !== 0) {
      console.log('fuck you');
      return;
    }

    event.stopPropagation();
    event.preventDefault();

    this.props.toggleUserDropdown();

  }

  render() {

    const { user } = this.props;

    const Notifications = (isLoggedIn) => {
      if (isLoggedIn) {
        return (
          <Link
            to="/user/notifications"
            className="btn notification">
            <i className="material-icons">
              notifications_none
            </i>
            <span className="notification-badge">N</span>
          </Link>
        );
      }
      return null;
    };

    const UserProfile = currentUser => (
      <div
        role="button"
        tabIndex={ 0 }
        className={ classnames('user-profile', {
          loggedIn: user.isLoggedIn
        }) }
        onMouseDown={ this.clickToToggle }
        onTouchEnd={ this.clickToToggle }
        onKeyDown={ () => {} }>
        {
          currentUser.isLoggedIn
            ? (
              <div className="user-info">
                <div className="user-avatar">
                  {
                    currentUser.avatar
                      ? (
                        <img
                          src={ currentUser.avatar }
                          alt={ currentUser.displayName } />
                      )
                      : (
                        <span>
                          { currentUser.displayName[0] }
                        </span>
                      )
                  }
                </div>
                <div className="user-username">
                  <span>{ currentUser.displayName }</span>
                  <span>{ currentUser.level || 'newbie' }</span>
                </div>
              </div>
            )
            : null
        }
        <a className="btn dropdown-toggle">
          <i className="material-icons dd-toggler">
            {
              this.props.layout.isUserDropdownOpen
                ? 'arrow_drop_up'
                : 'arrow_drop_down'
            }
          </i>
        </a>
      </div>
    );

    return (
      <div className="user-menu">
        { Notifications(user.isLoggedIn) }
        { UserProfile(user) }
      </div>
    );

  }

}

export default UserMenu;
