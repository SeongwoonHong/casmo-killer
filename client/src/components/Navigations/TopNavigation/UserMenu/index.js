import { connect } from 'react-redux';
import * as actions from 'actions';

import UserMenu from './UserMenu';

const mapStateToProps = (state) => {
  return {
    layout: state.layout,
    user: state.user
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    toggleUserDropdown: isOpen => dispatch(actions.toggleUserDropdown(isOpen)),
    openAuthModal: authType => dispatch(actions.openAuthModal(authType))
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(UserMenu);
