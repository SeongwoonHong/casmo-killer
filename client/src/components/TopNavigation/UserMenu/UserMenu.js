import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import classnames from 'classnames';
import './UserMenu.scss';

import UserMenuDropdown from './UserDropdown/UserMenuDropdown';

class UserMenu extends Component {

  render() {

    const { layout, user } = this.props;
    const isDeskTop = layout.breakPoint === 'lg' || layout.breakPoint === 'xl';

    const UserMenuButton = (isLoggedIn) => {
      if (isLoggedIn) {
        return (
          <a
            role="button"
            tabIndex={ 0 }
            className="btn teal darken-4"
            onBlur={ () => {
              if (layout.isUserMenuVisible) {
                this.props.toggleUserMenu();
              }
            } }
            onClick={ () => {
              this.props.toggleUserMenu();
            } }
            onKeyDown={ () => {
            } }>
            <i className="material-icons">
              {
                layout.isUserMenuVisible
                  ? 'person_outline'
                  : 'person'
              }
            </i>
          </a>
        );
      }
      return (
        <NavLink to="/login" className="btn teal darken-4">
          <i className="material-icons">person</i>
        </NavLink>
      );
    };

    return (
      <div className="user-menu">
        <NavLink
          to="/asdf"
          className={ classnames('btn teal darken-4', {
            hide: !user.isLoggedIn || !isDeskTop
          }) }>
          <i className="material-icons">
            notifications_none
          </i>
          <span className="notification-badge">
            N
          </span>
        </NavLink>
        { UserMenuButton(user.isLoggedIn) }
        {
          user.isLoggedIn && layout.isUserMenuVisible
            ? <UserMenuDropdown showNotification={ user.isLoggedIn && !isDeskTop } />
            : null
        }
      </div>
    );

  }

}

export default UserMenu;
