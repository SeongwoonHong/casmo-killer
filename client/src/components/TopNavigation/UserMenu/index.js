import { connect } from 'react-redux';
import UserMenu from './UserMenu';
import * as actions from '../../../actions';

const mapStateToProps = (state) => {
  return {
    layout: state.layout
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    toggleSearchForm: () => dispatch(actions.toggleSearchForm()),
    toggleMenu: () => dispatch(actions.toggleMenu())
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(UserMenu);
