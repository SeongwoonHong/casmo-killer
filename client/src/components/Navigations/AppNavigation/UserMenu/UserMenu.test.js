/* eslint react/jsx-boolean-value: 0 */
import React from 'react';
import { shallow } from 'enzyme';

import { UserMenu } from './UserMenu';

describe('<UserMenu />', () => {

  let component = null;

  beforeEach(() => {
    component = shallow(
      <UserMenu
        active={ true }
        user={ {
          isLoggedIn: false
        } }
        currentRoute="/"
        toggleUserMenu={ () => {} }
        logout={ () => {} } />
    );
  });

  it('renders the component consistently', () => {

    expect(component).toMatchSnapshot();

  });

});
