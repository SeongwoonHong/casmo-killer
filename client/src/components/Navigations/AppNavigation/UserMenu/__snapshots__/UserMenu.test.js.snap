// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserMenu /> renders the component consistently 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserMenu
    active={true}
    currentRoute="/"
    logout={[Function]}
    toggleUserMenu={[Function]}
    user={
        Object {
            "isLoggedIn": false,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "active": true,
      "children": Array [
        <UserDropdownBtn
          avatar=""
          displayName=""
          isLoggedIn={false}
          level=""
          toggleUserMenu={[Function]}
/>,
        <UserDropdown
          active={true}
          currentRoute="/"
          isLoggedIn={false}
          onLogout={[Function]}
          toggleUserMenu={[Function]}
/>,
      ],
      "className": "User-menu",
      "onClose": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "avatar": "",
          "displayName": "",
          "isLoggedIn": false,
          "level": "",
          "toggleUserMenu": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": true,
          "currentRoute": "/",
          "isLoggedIn": false,
          "onLogout": [Function],
          "toggleUserMenu": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "active": true,
        "children": Array [
          <UserDropdownBtn
            avatar=""
            displayName=""
            isLoggedIn={false}
            level=""
            toggleUserMenu={[Function]}
/>,
          <UserDropdown
            active={true}
            currentRoute="/"
            isLoggedIn={false}
            onLogout={[Function]}
            toggleUserMenu={[Function]}
/>,
        ],
        "className": "User-menu",
        "onClose": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "avatar": "",
            "displayName": "",
            "isLoggedIn": false,
            "level": "",
            "toggleUserMenu": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": true,
            "currentRoute": "/",
            "isLoggedIn": false,
            "onLogout": [Function],
            "toggleUserMenu": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
